draft-ietf-gnap-core-protocol.md

```diff
@@ -2368,108 +2370,199 @@ unknown_request
-cases where the end-user and RO are separate parties and the authorization
-happens asynchronously.
+The client instance MUST NOT modify the URI when launching it,
+in particular the client instance MUST NOT add any parameters to the URI.
+The user code URI MUST be reachable from the end user's browser, though
+the URI is usually be opened on a separate device from the client instance
+itself. The URI MUST be accessible from an HTTP GET
+request and MUST be protected by HTTPS or equivalent means.
 
+### Interaction through an Application URI {#interaction-app}
 
-## Interaction through an Application URI {#interaction-app}
+When the client instance is directed to launch an application through the
+["app"](#response-interact-app) mode, the client launches the
+URL as appropriate to the system. The means by which the AS and the
+launched application communicate with each other and perform any
+of the required actions are out of scope for this specification.
 
-When the client instance successfully launches an application through the
-["app" mode](#response-interact-app), the AS
-interacts with the RO through that application to authenticate the
-user as the RO and gather their consent. The details of this
-interaction are out of scope for this specification.
+## Post-Interaction Completion {#interaction-finish}
 
-## Post-Interaction Completion {#interaction-finalize}
-
-Upon completing an interaction with the RO, if a ["callback"](#response-interact-callback) mode is
-available with the current request, the AS MUST follow the appropriate
-method at the end of interaction to allow the client instance to continue. If
-this mode is not available, the AS SHOULD instruct the RO to
-return to their client instance upon completion. Note that these steps
-still take place in most error cases, such as when the RO has denied
-access. This pattern allows the client instance to potentially recover from the error
-state without restarting the request from scratch by modifying its
-request or providing additional information directly to the AS.
+If an interaction ["finish"](#response-interact-finish) method is
+associated with the current request, the AS MUST follow the appropriate
+method at upon completion of interaction in order to signal the client instance to continue.
+If a finish method is not available, the AS SHOULD instruct the RO to
+return to the client instance upon completion.
 
 The AS MUST create an interaction reference and associate that
 reference with the current interaction and the underlying pending
-request. This value MUST be sufficiently random so as not to be
+request. This interaction reference value MUST be sufficiently random so as not to be
 guessable by an attacker. The interaction reference MUST be
-one-time-use.
+one-time-use to prevent interception and replay attacks.
 
 The AS MUST calculate a hash value based on the client instance and AS nonces and the
```

Comment:
that's a minor comment, but nonce in british english doesn't mean at all the same as in American ...

Reply:
I haven't seen an alternative embraced by the community yet, but I'd be happy for us to change if a good suggestion is made.